src/transforms/pipelines/mod.rs

```diff
@@ -0,0 +1,200 @@
+mod expander;
+mod router;
+
+use crate::config::{
+    DataType, ExpandType, GenerateConfig, TransformConfig, TransformContext, TransformDescription,
+};
+use crate::transforms::Transform;
+use indexmap::IndexMap;
+use serde::{Deserialize, Serialize};
+
+const SAMPLE_CONFIG: &str = indoc::indoc! { r#"
+[logs]
+order = ["foo", "bar"]
+
+[logs.pipelines.foo]
+name = "foo pipeline"
+
+[[logs.pipelines.foo.transforms]]
+type = "filter"
+condition = ""
+
+[[logs.pipelines.foo.transforms]]
+type = "filter"
+condition = ""
+
+[logs.pipelines.bar]
+name = "bar pipeline"
+
+[[logs.pipelines.bar.transforms]]
+type = "filter"
+condition = ""
+"# };
+
+inventory::submit! {
+    TransformDescription::new::<PipelinesConfig>("pipelines")
+}
+
+#[derive(Debug, Default, Deserialize, Serialize)]
+// #[serde(deny_unknown_fields)]
+pub struct PipelineConfig {
+    name: String,
+    transforms: Vec<Box<dyn TransformConfig>>,
+}
```

Comment:
What is the relationship of this type to its pluralized sibling, `PipelinesConfig`?

Reply:
Nope, I see where it's complex.
The `pipelines` transform contains a set of pipelines.
In each pipeline, you have some transforms. So this is really `PipelineConfig` because it's the configuration of a single pipeline.